packages/cli/src/chat/chat-service.ts

```diff
@@ -0,0 +1,298 @@
+
+					connection.send(textMessage, () => {
+						closeConnection(connection);
+					});
+
+					session.isProcessing = false;
+					return;
+				}
+
+				session.isProcessing = false;
+			} catch (error) {
+				if (session) session.isProcessing = false;
+				this.logger.error(
+					`Error sending message to chat in session ${sessionKey}: ${(error as Error).message}`,
+				);
+			}
+		};
+	}
+
+	private incomingMessageHandler(sessionKey: string) {
+		return async (data: RawData) => {
+			try {
+				const session = this.sessions.get(sessionKey);
+
+				if (!session) return;
+
+				const message = data.toString();
+
+				if (message === 'n8n|heartbeat-ack') {
+					session.lastHeartbeat = Date.now();
+					return;
+				}
+
+				const executionId = session.executionId;
+
+				await this.resumeExecution(executionId, this.processIncomingData(data), sessionKey);
+				session.waitingNodeName = undefined;
+			} catch (error) {
+				this.logger.error(
+					`Error processing message from chat in session ${sessionKey}: ${(error as Error).message}`,
+				);
+			}
+		};
+	}
+
+	private async resumeExecution(executionId: string, message: ChatMessage, sessionKey: string) {
+		const execution = await this.getExecution(executionId, sessionKey);
+		if (!execution) return;
+		await this.executionManager.runWorkflow(execution, message);
+	}
+
+	private async getExecution(executionId: string, sessionKey: string) {
+		const execution = await this.executionManager.findExecution(executionId);
+
+		if (!execution || ['error', 'canceled', 'crashed'].includes(execution.status)) {
+			const session = this.sessions.get(sessionKey);
+
+			if (!session) return null;
+
+			session.connection.terminate();
+			clearInterval(session.intervalId);
+			this.sessions.delete(sessionKey);
+			return null;
+		}
+
+		if (execution.status === 'running') return null;
+
+		return execution;
+	}
+
+	private processIncomingData(data: RawData) {
+		const buffer = Array.isArray(data)
+			? Buffer.concat(data.map((chunk) => Buffer.from(chunk)))
+			: Buffer.from(data);
+
+		const message = jsonParse<ChatMessage>(buffer.toString('utf8'));
+
+		if (message.files) {
+			message.files = message.files.map((file) => ({
+				...file,
+				data: file.data.includes('base64,') ? file.data.split('base64,')[1] : file.data,
+			}));
+		}
+
+		return message;
+	}
+
+	private async checkHeartbeats() {
+		try {
+			const now = Date.now();
+			const disconnected: string[] = [];
+
+			for (const [key, session] of this.sessions.entries()) {
+				if (!session) continue;
+
+				const timeSinceLastHeartbeat = now - (session.lastHeartbeat || 0);
+
+				if (timeSinceLastHeartbeat > HEARTBEAT_TIMEOUT) {
+					await this.executionManager.cancelExecution(session.executionId);
+					session.connection.terminate();
+					clearInterval(session.intervalId);
+					disconnected.push(key);
+				} else {
+					try {
+						session.connection.send('n8n|heartbeat');
+					} catch (error) {
+						disconnected.push(key);
```

Comment:
If we enter here, this session's interval and connection never get cleared, they are only removed from the session map?